### PR TITLE
Improve abstract api features error messages

### PR DESCRIPTION
**Pull request by Buckram123**
Fixes ABS-210

Most of API raw queries responses was already handled. Do we want to do extra-handling for smart queries as well? 
i.e. we can add new error type to AbstractSdkError:
```rust
    #[error("API smart query for {object} failed: {error}")]
    ApiSmartQuery {
        object: String, // For example "ModuleRegistry",
        error: cosmwasm_std::StdError,
    },
```  
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-sdk/src/apis/app.rs [hunk 1]
```diff
@@ -67,26 +67,26 @@ impl<'a, T: AppInterface> Apps<'a, T> {
...
-        Ok(wasm_execute(app_address, &app_query, vec![])?.into())
+        Ok(wasm_execute(app_address, &base_msg, vec![])?.into())
     }
 
     /// Smart query an app
     pub fn query<Q: Serialize, R: DeserializeOwned>(
         &self,
         app_id: ModuleId,
-        message: impl Into<msg::QueryMsg<Q>>,
+        query: impl Into<msg::QueryMsg<Q>>,
```
Comment (CyberHoward) [inline framework/packages/abstract-sdk/src/apis/app.rs:82]: lol, good spot!!

File: framework/packages/abstract-sdk/src/apis/version_registry.rs [hunk 2]
```diff
@@ -95,29 +92,34 @@ impl<'a, T: ModuleRegistryInterface> ModuleRegistry<'a, T> {
...
     ) -> AbstractSdkResult<Vec<ModuleResponse>> {
         let registry_addr = self.base.abstract_registry(self.deps)?.address;
-        let ModulesResponse { modules } = self.deps.querier.query(&wasm_smart_query(
-            registry_addr.into_string(),
-            &QueryMsg::Modules { infos },
-        )?)?;
+        let ModulesResponse { modules } = self
+            .deps
+            .querier
+            .query_wasm_smart(registry_addr.into_string(), &QueryMsg::Modules { infos })?;
```
Comment (CyberHoward) [inline framework/packages/abstract-sdk/src/apis/version_registry.rs:98]: Could these queries get their own Error message? Should help in debugging if a module is missing.

Comment (CyberHoward) [reply framework/packages/abstract-sdk/src/apis/version_registry.rs:98]: Nevermind, the smart query should return a good error anyway!

**Comment by CyberHoward** [comment]
> Most of API raw queries responses was already handled. Do we want to do extra-handling for smart queries as well? i.e. we can add new error type to AbstractSdkError:
> 
> ```rust
>     #[error("API smart query for {object} failed: {error}")]
>     ApiSmartQuery {
>         object: String, // For example "ModuleRegistry",
>         error: cosmwasm_std::StdError,
>     },
> ```

This could be very useful, especially if we add an `in: String` as well, so: 

```rust
#[error("API smart query for {object} failed in {in}: {error}")]
     ApiSmartQuery {
         object: String, // For example "ModuleRegistry",
         in: String,
         error: cosmwasm_std::StdError,
     },
```

We could fill the `in` field with `base.module_id` if we add the `ModuleIdentification` as a trait bound?

**Comment by Buckram123** [comment]
> > Most of API raw queries responses was already handled. Do we want to do extra-handling for smart queries as well? i.e. we can add new error type to AbstractSdkError:
> This could be very useful, especially if we add an `in: String` as well, so:
> 
> ```rust
> #[error("API smart query for {object} failed in {in}: {error}")]
>      ApiSmartQuery {
>          object: String, // For example "ModuleRegistry",
>          in: String,
>          error: cosmwasm_std::StdError,
>      },
> ```
> 
> We could fill the `in` field with `self.base.module_id` if we add the `ModuleIdentification` as a trait bound?

Yeah that's really nice for debugging! Could it have any significant downsides?

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**